aibtc-v1/crews/contract_code_generator.py

```diff
@@ -0,0 +1,194 @@
+import os
+import subprocess
+from crewai import Agent, Task, Crew, Process
+from crewai_tools import tool
+from langchain_openai import ChatOpenAI
+from langchain_ollama import OllamaLLM
+from dotenv import load_dotenv
+import streamlit as st
+
+# Load environment variables
+load_dotenv()
+
+
+@tool("Clarinet")
+def runClarinet(project_name: str, contract_name: str, contract_code: str) -> str:
+    """
+    Create a new Clarinet project, add a new contract, and check its syntax.
+
+    This tool creates a new Clarinet project, adds a new contract to it,
+    writes the provided contract code, and checks its syntax.
+
+    Args:
+        project_name (str): The name of the Clarinet project to be created.
+        contract_name (str): The name of the contract to be created.
+        contract_code (str): The code to be written into the new contract file.
+
+    Returns:
+        str: A message indicating the result of the operations.
+    """
+    initial_dir = os.getcwd()
+    try:
+        # Create a new Clarinet project
+        subprocess.run(["clarinet", "new", project_name], check=True)
+        print(f"Created new Clarinet project: {project_name}")
+
+        # Change directory to the project folder
+        os.chdir(project_name)
+
+        # Add a new contract
+        subprocess.run(["clarinet", "contract", "new",
+                       contract_name], check=True)
+        print(f"Added new contract: {contract_name}")
+
+        # Write the contract code to the contract file
+        contract_file_path = os.path.join("contracts", f"{contract_name}.clar")
+        with open(contract_file_path, "w") as contract_file:
+            contract_file.write(contract_code)
+        print(f"Wrote code to {contract_file_path}")
+
+        # Check the syntax of the contract
+        result = subprocess.run(
```

Comment:
What happens if the check fails and the agent tries again? They'd end up re-running both the `clarinet new` and `clarinet contract new` commands a second time.

Maybe those should be separate tools?
- create_clarinet_project
- create_new_smart_contract
- check_smart_contract_syntax